1701/h/web 2780 error logging

rahulk1000 opened the pull request.

InvokeJLa commented on test/unit/exception_handling_test.rb:
```diff
@@ -135,11 +148,37 @@ def log_error_callback_with_failure(data, ex)
+      end
       assert_nothing_raised { ExceptionHandling.ensure_safe("mooo") { raise "Some BS" } }
-      assert_equal 0, @fail_count
+      assert log_info_messages.find { |message| message =~ /Unable to execute custom log_error callback/ }
```
Since `@fail_count` is only used once in the test, can we move it out of the setup block and into the method that uses it? (line 139)

rahulk1000 replied:
done

InvokeJLa commented on test/unit/exception_handling_test.rb:
```diff
@@ -135,11 +148,37 @@ def log_error_callback_with_failure(data, ex)
       ExceptionHandling.post_log_error_hook = nil
     end
 
+    should "handle nil message exceptions resulting from the log_error hook" do
```
Not sure if it's worth it, but this test includes some similar setup code as the test above it. Maybe combine it under a context + setup block?

rahulk1000 replied:
don't think it's worth it

InvokeJLa commented:
@rahulk1000 Done.

rahulk1000 commented:
@InvokeJLa replied to / addressed your CR comments. Additionally, refactored some code in exception_info.rb and added some missing tests in exception_info_test.rb

InvokeJLa commented on test/unit/exception_handling/exception_info_test.rb:
```diff
@@ -261,6 +279,23 @@ class ExceptionInfoTest < ActiveSupport::TestCase
+        ExceptionHandling.custom_data_hook = lambda do |data|
+          exception_with_nil_message = RuntimeError.new(nil)
+          stub(exception_with_nil_message).message { nil }
+          raise exception_with_nil_message
```
These 3 lines are used in `exception_handling_test` as well for custom data. Helper method for it?

rahulk1000 replied:
done

InvokeJLa commented on lib/exception_handling/version.rb:
```diff
@@ -1,3 +1,3 @@
 module ExceptionHandling
-  VERSION = "1.2.6"
+  VERSION = "1.2.7"
```
Discussed in person: Make sure the gem update in web works, including engines.

InvokeJLa commented:
@rahulk1000 Done.

mikeweaver commented:
LGTM

InvokeJLa commented:
👍

mikeweaver commented:
LGTM
